src/structures/ModalSubmitInteraction.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const BaseMessageComponent = require('./BaseMessageComponent');
+const Interaction = require('./Interaction');
+const InteractionResponses = require('./interfaces/InteractionResponses');
+
+class ModalSubmitInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The custom id of the modal.
+     * @type {string}
+     */
+    this.customId = data.data.custom_id;
+
+    /**
+     * The inputs within the modal
+     * @type {Array<MessageActionRow<MessageActionRowComponent>>}
+     */
+    this.components = data.data.components?.map(c => BaseMessageComponent.create(c, this.client)) ?? [];
+  }
+
+  // These are here only for documentation purposes - they are implemented by InteractionResponses
+  /* eslint-disable no-empty-function */
+  deferReply() {}
+  reply() {}
+  fetchReply() {}
+  editReply() {}
+  deleteReply() {}
+  followUp() {}
```

Comment:
shouldn't update() be documented here?

Reply:
How you'll update a model?

Reply:
Modal Submit interactions do not support update, deferUpdate, or replying with another modal.

Reply:
I'm confused, this code works fine, doesn't that imply that model submit interactions should have an update() method? or am I missing something
```js
this.client.on('interactionCreate', async (i) => {
			if (i.isModalSubmit()) {
				//@ts-ignore
				await this.client.api.interactions(i.id, i.token).callback.post({
					data: {
						type: 7, //update original msg
						data: { content: 'Updated' }
					}
				});
			}
		});
```

Reply:
So this is a bit confusing, mainly because modals can be triggered by any interaction type. This means that they can be triggered but a button interaction on a message. So technically in those situations you can update the message from where the interaction was sent.